### ISSUE TITLE
Most commands say my version is outdated (3.1.0), but the upgrade command does not increase the OnePub version

### ISSUE THREAD

**Issue by gabrielbertollo** [onepub-dev/onepub#528]
**Describe the bug**
Most of the commands I run are displaying the following message, and are not allowing me to proceed:
```
OnePub version: 3.1.0 

The server's major version "4" does not match your onepub version.

Please upgrade onepub by running:
dart pub global activate onepub
```
I already tried running the suggested command to upgrade, to no avail. This is also happening in my CI/CD environments, so I'm unable to deploy my apps.

This is the output of the `dart pub global activate onepub` command:

```
Package onepub is currently active at version 3.1.0.
The package onepub is already activated at newest available version.
To recompile executables, first run `dart pub global deactivate onepub`.
Installed executables onepub and opub.
Activated onepub 3.1.0.
```

**To Reproduce**
1. Go to terminal
2. Run any OnePub command, such as `onepub login` or `onepub export --user [email]`
3. See error

**Expected behavior**
As I see it, the command `dart pub global activate onepub` should get the latest version of OnePub.

** Problems with CLI tooling **
OnePub doctor command output:

```
OnePub version: 3.1.0 

Platform
Dart version:   2.19.1                             
Dart path:      /Users/gabrielb/development/flutter

OS:             macos                              
OS version:     Version 13.1 (Build 22C65)         

Locale:         en-BR                              
Path separator: /                                  


URLs
Web site:       https://onepub.dev/                
API endpoint:   https://onepub.dev/api/xpdpsdavuh/ 

Environment
Pub Cache:      /<HOME>/.pub-cache                 
PUB_CACHE:      not set and not used.              
PATH:
                /opt/homebrew/bin                   
                /opt/homebrew/sbin                  
                /usr/local/bin                      
                /System/Cryptexes/App/usr/bin       
                /usr/bin                            
                /bin                                
                /usr/sbin                           
                /sbin                               
                /Library/Apple/usr/bin              
                /opt/homebrew/bin                   
                /opt/homebrew/sbin                  
                /<HOME>/development/flutter/bin     
                /<HOME>/.pub-cache/bin              
                /<HOME>/development/flutter/bin     
                /<HOME>/.pub-cache/bin              

Shell Settings
$SHELL:         /bin/zsh                           
Detected:       zsh                                
Start script:   /<HOME>/.zshrc                     

Repository tokens
https://onepub.dev/api/advdavuhno/
https://onepub.dev/api/xpdpsdavuh/

Status
Logged In:      true                               
Active Member:  code@prefeituradofuturo.com.br     
Organisation:   code@prefeituradofuturo.com.br     
checking status...  
OnePub: status healthy.

Server Version: 4.13.1
**************************************** ERROR ****************************************
The OnePub Server version does not match your onepub version.
Please upgraded onepub by running:
dart pub global activate onepub
**************************************** ERROR ****************************************
```

**Comment by bsutton** [comment]
I believe we understand the problem and we should be able to sort this out
today.

Will update you as we proceed.

**Comment by bsutton** [comment]
Fyi: the quick fix is to update to dart 3.x and the update onepub (if that
is viable for you)

But we are working in a fix for 2.19

**Comment by bsutton** [comment]
Gabriel,
we have resolved the issue.

Please run 'dart pub global activate onepub.

You should see v 3.2.0 is activated.

Then run onepub doctor to ensure that it is working as expected.

Apologies for the inconvenience.

I will close this issue as resolved but feel free to re-open it if you have any issues.

**Comment by gabrielbertollo** [comment]
Hey there Brett,

Is this fix working for Dart 2.19 or do I have to upgrade?

Currently, due to some package limitations, I cannot upgrade, and in Dart 2.19 the commands are still returning the same output even with OnePub at 3.2.0.

```
OnePub version: 3.2.0 

The server's major version "4" does not match your onepub version.

Please upgrade onepub by running:
dart pub global activate onepub
```

**Comment by gabrielbertollo** [comment]
This is the current OnePub doctor output:
```
OnePub version: 3.2.0 

Platform
Dart version:   2.19.1                             
Dart path:      /Users/gabrielb/development/flutter

OS:             macos                              
OS version:     Version 13.1 (Build 22C65)         

Locale:         en-BR                              
Path separator: /                                  


URLs
Web site:       https://onepub.dev/                
API endpoint:   https://onepub.dev/api/xpdpsdavuh/ 

Environment
Pub Cache:      /<HOME>/.pub-cache                 
PUB_CACHE:      not set and not used.              
PATH:
                /opt/homebrew/bin                   
                /opt/homebrew/sbin                  
                /usr/local/bin                      
                /System/Cryptexes/App/usr/bin       
                /usr/bin                            
                /bin                                
                /usr/sbin                           
                /sbin                               
                /Library/Apple/usr/bin              
                /opt/homebrew/bin                   
                /opt/homebrew/sbin                  
                /<HOME>/development/flutter/bin     
                /<HOME>/.pub-cache/bin              
                /<HOME>/development/flutter/bin     
                /<HOME>/.pub-cache/bin              

Shell Settings
$SHELL:         /bin/zsh                           
Detected:       zsh                                
Start script:   /<HOME>/.zshrc                     

Repository tokens
https://onepub.dev/api/advdavuhno/
https://onepub.dev/api/xpdpsdavuh/

Status
Logged In:      true                               
Active Member:  code@prefeituradofuturo.com.br     
Organisation:   code@prefeituradofuturo.com.br     
checking status...  
OnePub: status healthy.

Server Version: 4.14.2
```

When trying to run onepub export or onepub login:
```
OnePub version: 3.2.0 

The server's major version "4" does not match your onepub version.

Please upgrade onepub by running:
dart pub global activate onepub
```

**Comment by bsutton** [comment]
Gabriel,
we have just released 3.2.1.
Dart 2.19 is supported. We currently support all versions of dart from 2.12.

We missed a couple of scenarios and have now consolidated the tests.

Any problems let me know.
